src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs

```diff
@@ -10,17 +10,14 @@ namespace Microsoft.Extensions.Logging.Console
 {
     public class ConsoleLogger : ILogger
     {
-        // Writing to console is not an atomic operation in the current implementation and since multiple logger
-        // instances are created with a different name. Also since Console is global, using a static lock is fine.
-        private static readonly object _lock = new object();
         private static readonly string _loglevelPadding = ": ";
         private static readonly string _messagePadding;
         private static readonly string _newLineWithMessagePadding;
 
         // ConsoleColor does not have a value to specify the 'Default' color
         private readonly ConsoleColor? DefaultConsoleColor = null;
 
-        private IConsole _console;
+        private readonly ConsoleLoggerProcessor _queueProcessor = new ConsoleLoggerProcessor();
```

Comment:
This spins up a thread per category, we need to move this to the provider so there's a single thread and remeasure.